QgisModelBaker/libili2db/iliexecutable.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+"""
+/***************************************************************************
+    begin                :    09/09/23
+    git sha              :    :%H$
+    copyright            :    (C) 2020 by Yesid Polania
+    email                :    yesidpol.3@gmail.com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+import functools
+import locale
+import re
+from qgis.PyQt.QtCore import QObject, pyqtSignal, QProcess, QEventLoop
+from abc import ABCMeta, abstractmethod
+
+from QgisModelBaker.libili2db.ili2dbutils import get_java_path, get_ili2db_bin, JavaNotFoundError
+from QgisModelBaker.libqgsprojectgen.db_factory.db_simple_factory import DbSimpleFactory
+
+
+class AbstractQObjectMeta(ABCMeta, type(QObject)):
+    pass
+
+
+class IliExecutable(QObject, metaclass=AbstractQObjectMeta):
+    SUCCESS = 0
+    # TODO: Insert more codes?
+    ERROR = 1000
+    ILI2DB_NOT_FOUND = 1001
+
+    stdout = pyqtSignal(str)
+    stderr = pyqtSignal(str)
+    process_started = pyqtSignal(str)
+    process_finished = pyqtSignal(int, int)
+
+    __done_pattern = None
+    __result = None
+
+    def __init__(self, parent=None):
+        QObject.__init__(self, parent)
+        self.filename = None
+        self.tool = None
+        self.configuration = self._create_config()
+        self.encoding = locale.getlocale()[1]
```

Comment:
```suggestion
        _, self.encoding = locale.getlocale()
```

I think that's the nice way to access a tuple. It's well possible your python skills are better than mine though.